client/consul/consul.go

```diff
@@ -78,7 +92,14 @@ type ConsulClientFunc func(config *config.ConsulConfig, logger hclog.Logger) (Cl
 // NewConsulClientFactory returns a ConsulClientFunc that closes over the
 // partition
 func NewConsulClientFactory(node *structs.Node) ConsulClientFunc {
+
+	// these node values will be evaluated at the time we create the hooks, so
+	// we don't need to worry about them changing out from under us
 	partition := node.Attributes["consul.partition"]
+	preflightCheckTimeout := durationFromMeta(
+		node, "consul.token_preflight_check.timeout", time.Second*10)
+	preflightCheckBaseInterval := durationFromMeta(
+		node, "consul.token_preflight_check.base", time.Millisecond*500)
```

Comment:
Hm, I'm not super familiar with this code, but it appears `NewConsulClientFactory` will be called  from `NewClient` and therefore the timeout and interval node meta will be captured only during agent startup. Don't the durationFromMeta calls need to happen inside the closure and use `client.Node()` instead of a captured Node?

Reply:
That's a good catch. For partitions, we don't expect that to change because it's part of the fingerprint we get from the agent. But metadata can be changed via the `node meta` command so yeah we'll need to move this into the closure.

Reply:
Fixed! (And rebased on main) Also:

> I'm not super familiar with this code
The various dependency injection tricks we're doing here to avoid circular references doesn't help. This code all got really janky during the big WI + Consul/Vault refresh. Once we remove the legacy workflow we should be able to refactor this in the process to be a little more comprehensible.